src/Bedrock.Framework.Experimental/Protocols/Memcached/MemcachedResponse.cs

```diff
@@ -0,0 +1,57 @@
+﻿using Bedrock.Framework.Infrastructure;
+using System;
+using System.Buffers;
+using System.Buffers.Binary;
+using System.Collections.Generic;
+using System.Text;
+using static Bedrock.Framework.Experimental.Protocols.Memcached.Enums;
+
+namespace Bedrock.Framework.Experimental.Protocols.Memcached
+{
+    public class MemcachedResponse
+    {
+        public ReadOnlyMemory<byte> Data { get; private set; }
+        public TypeCode Flags { get; set; }
+        public MemcachedResponseHeader Header { get; private set; }
+
+        public void ReadHeader(ReadOnlySpan<byte> buffer)
+        {
+
+            if (buffer[0] != MemcachedResponseHeader.Magic)
+            {
+                throw new ArgumentException("Magic mismatch");
+            }
+
+            this.Header = new MemcachedResponseHeader()
+            {
+                Opcode = (Opcode)buffer[1],
+                KeyLength = BinaryPrimitives.ReadUInt16BigEndian(buffer.Slice(2)),
+                ExtraLength = buffer[4],
+                DataType = buffer[5],
+                ResponseStatus = (ResponseStatus)BinaryPrimitives.ReadUInt16BigEndian(buffer.Slice(6)),
+                TotalBodyLength = BinaryPrimitives.ReadUInt32BigEndian(buffer.Slice(8)),
+                Opaque = BinaryPrimitives.ReadUInt32BigEndian(buffer.Slice(12)),
+                Cas = BinaryPrimitives.ReadUInt64BigEndian(buffer.Slice(16)),
+            };
+        }
+
+        public void ReadBody(ReadOnlySequence<byte> sequence)
+        {
+            if (sequence.Length == 0)
+            {
+                return;
+            }
+                
+            if (sequence.IsSingleSegment)
+            {
+                Flags = (TypeCode)BinaryPrimitives.ReadUInt32BigEndian(sequence.First.Span);
+                Data = sequence.First.Slice(Header.KeyLength + Header.ExtraLength);
+            }
+            else
+            { 
+                Flags = (TypeCode)BinaryPrimitives.ReadUInt32BigEndian(sequence.Slice(0).FirstSpan);
```

Comment:
This assumes the FirstSpan has at least 4 bytes. While reasonable in general it's not "correct".

Reply:
i think i'll remove flags info since your comment about api which should just support byte.
What do you think about it ?

Reply:
The fix here is easy enough:

```C#
Span<byte> buffer = stackalloc[4];
sequence.Slice(0, 4).CopyTo(buffer);
Flags = (TypeCode)BinaryPrimitives.ReadUInt32BigEndian(buffer);
```

Reply:
You actually want to slice before doing the IsSingleSegment check. That way you're checking a very small buffer vs potentially a buffer with multiple messages.

Reply:
i'm not sure to understand. This way ? [https://github.com/davidfowl/BedrockFramework/pull/84/commits/cdccfb1a03af7ccfbfca474a143ffe9ddece486d](https://github.com/davidfowl/BedrockFramework/pull/84/commits/cdccfb1a03af7ccfbfca474a143ffe9ddece486d)